[CI] 自动测试失败 - 2026-03-01 03:30

## CI Monitor 自动检测到测试失败

**触发提交:**
- 8c9309f Merge pull request #407 from hs3180/refactor/issue-283-reflection-controller
- 2ffdf41 refactor(task): replace DialogueOrchestrator with ReflectionController
- 73340bd Merge pull request #410 from hs3180/test/issue-397-coverage-fix
- 54e23db test: add message-adapter tests and fix TypeScript error

## 测试结果摘要

| 测试类型 | 结果 |
|---------|------|
| 类型检查 (type-check) | ✅ 通过 |
| Lint 检查 | ✅ 通过 (62 warnings) |
| 单元测试 | ✅ 通过 (67 files, 1221 tests) |
| 构建 (build) | ✅ 通过 |
| 集成测试 | ❌ **失败** |

## 失败详情

### 集成测试失败

**失败的测试:**
1. **Use Case 1 - Test 3: Custom chatId preservation** - HTTP 000 (连接失败)
2. **Use Case 2 - Test 3: File listing task** - HTTP 000 (连接失败)

**根本原因:**
服务器日志显示端口冲突错误:
```
Error: listen EADDRINUSE: address already in use 0.0.0.0:3099
Error: listen EADDRINUSE: address already in use 127.0.0.1:3001
```

当测试脚本尝试启动新的服务器实例时，前一个测试的服务器可能没有正确清理，导致端口仍被占用。RestChannel 启动失败后，后续请求返回 HTTP 000。

**日志片段:**
```
[ERROR] [RestChannel] Failed to start RestChannel
    err: {
      "type": "Error",
      "message": "listen EADDRINUSE: address already in use 0.0.0.0:3099",
      ...
    }
[ERROR] [PrimaryNode] Failed to start channel
    channelId: "rest"
```

## 环境信息
- Node.js: v24.8.0
- 测试时间: 2026-03-01 03:26
- 测试配置: disclaude.config.test.yaml

## 建议修复方向
1. 检查测试脚本 `tests/integration/run-all-tests.sh` 中的服务器清理逻辑
2. 确保在启动新服务器前正确终止前一个服务器进程
3. 考虑添加端口等待/重试逻辑

---
🤖 此 Issue 由 CI Monitor 自动生成

--- a/tests/integration/common.sh
+++ b/tests/integration/common.sh
@@ -97,10 +97,78 @@ log_debug() {
 # Server Management Functions
 # =============================================================================
 
+# Check if a port is in use
+# Returns: 0 if port is in use, 1 if port is free
+is_port_in_use() {
+    local port="$1"
+    if command -v lsof &> /dev/null; then
+        lsof -i:"$port" -sTCP:LISTEN > /dev/null 2>&1
+    elif command -v ss &> /dev/null; then
+        ss -tln | grep -q ":${port} "
+    elif command -v netstat &> /dev/null; then
+        netstat -tln | grep -q ":${port} "
+    else
+        # Fallback: try to connect
+        curl -s "http://${HOST}:${port}/api/health" > /dev/null 2>&1
+    fi
+}
+
+# Check if server is already running on the target port
+# Returns: 0 if server is running and healthy, 1 otherwise
+is_server_running() {
+    curl -s "${API_URL}/api/health" > /dev/null 2>&1
+}
+
+# Wait for port to be released
+# Returns: 0 if port is released, 1 if timeout
+wait_for_port_release() {
+    local port="$1"
+    local max_retries="${2:-10}"
+    local retry=0
+
+    while [ $retry -lt $max_retries ]; do
+        if ! is_port_in_use "$port"; then
+            log_debug "Port $port is now free"
+            return 0
+        fi
+        sleep 1
+        retry=$((retry + 1))
+        log_debug "Waiting for port $port to be released... ($retry/$max_retries)"
+    done
+
+    log_warn "Port $port still in use after ${max_retries} seconds"
+    return 1
+}
+
 # Start the test server
 # Returns: 0 on success, 1 on failure
 start_server() {
     log_info "Starting test server on port ${REST_PORT}..."
+
+    # Check if server is already running and healthy
+    if is_server_running; then
+        log_info "Server already running on port ${REST_PORT}, reusing existing server"
+        SERVER_PID=""
+        return 0
+    fi
+
+    # Wait for port to be released if it's in use but server is not healthy
+    if is_port_in_use "$REST_PORT"; then
+        log_warn "Port ${REST_PORT} is in use but server is not healthy, waiting for release..."
+        if ! wait_for_port_release "$REST_PORT" 15; then
+            log_error "Port ${REST_PORT} is still in use, cannot start server"
+            # Try to kill any process using the port
+            if command -v lsof &> /dev/null; then
+                local pid_using_port
+                pid_using_port=$(lsof -t -i:"$REST_PORT" 2>/dev/null | head -1)
+                if [ -n "$pid_using_port" ]; then
+                    log_warn "Killing process $pid_using_port using port ${REST_PORT}"
+                    kill -9 "$pid_using_port" 2>/dev/null || true
+                    sleep 2
+                fi
+            fi
+        fi
+    fi
 
     cd "$PROJECT_ROOT"
 
@@ -142,6 +210,9 @@ stop_server() {
         kill $SERVER_PID 2>/dev/null || true
         wait $SERVER_PID 2>/dev/null || true
         SERVER_PID=""
+
+        # Wait for port to be released
+        wait_for_port_release "$REST_PORT" 10 || true
     fi
 }
 

